mne/channels/channels.py

```diff
@@ -228,6 +229,70 @@ def get_channel_types(self, picks=None, unique=False, only_data_chs=False):
         return _get_channel_types(self.info, picks=picks, unique=unique,
                                   only_data_chs=only_data_chs)
 
+    def get_ch_positions(self, picks=None):
+        """Get a dictionary of channel positions for each channel.
+
+        Parameters
+        ----------
+        picks : str | list | slice | None
+            None gets good data indices.
+        Returns
+        -------
+        ch_positions : dict
+            A dictionary with keys as the ch_names and values as
+             corresponding list of xyz coordinates.
+        """
+        # get the channel positions
+        picks = _picks_to_idx(self.info, picks)
+        chs = self.info['chs']
+        ch_names = self.info['ch_names']
+        pos = np.array([chs[k]['loc'][:3] for k in picks])
+        return dict(zip(ch_names, pos))
+
+    def get_montage(self):
+        """Get a DigMontage from instance.
+
+        Returns
+        -------
+        %(montage)s
+        """
+        from ..channels.montage import make_dig_montage
+        if self.info['dig'] is None:
+            return None
+
+        # get channel positions
+        ch_pos = self.get_ch_positions()
+
+        # get coordinate frame
+        coord_frame_int = self.info['dig'][0]['coord_frame']
+        coord_frame = _frame_to_str[coord_frame_int]
+
+        # extract landmark coords
+        landmark_coords = _extract_landmarks(self.info['dig'])
+
+        # create montage and return it
+        montage = make_dig_montage(
+            ch_pos=ch_pos,
+            coord_frame=coord_frame,
+            **landmark_coords
```

Comment:
you're omitting all the dig points that are not EEG channels or fids.

have a look at function `_foo_get_data_from_dig` that should probably be renamed if you start using it.  Don't copy the code blindly as it would make a code duplication